### PR TITLE
[RFR] Docs restructured

### PR DESCRIPTION
**Pull request by jan-zmeskal**
So what's the point of this PR? Let's go through it:
* Documentation is now split between source and build, build is ignored by version control
* I disabled generation of API docs, I don't believe we're there yet
* Usage information is generated automatically based on `disruption_generator --help` output
* Fake logger dev utility is documented

I generally think that the current structure is cleaner and it is easier for future contributors to just jump in and extend the documentation. Let me know what do you think.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -63,7 +63,8 @@ instance/
 .scrapy
 
 # Sphinx documentation
-docs/_build/
+docs/build/
+docs/source/usage/help_output.inc
```
Comment (jan-zmeskal) [inline .gitignore]: This is generated dynamically

File: Makefile [hunk 2]
```diff
@@ -65,15 +65,13 @@ coverage: ## check code coverage quickly with the default Python
 	coverage html
 	$(BROWSER) htmlcov/index.html
 
-docs: ## generate Sphinx HTML documentation, including API docs
-	rm -f docs/disruption_generator.rst
-	rm -f docs/modules.rst
-	sphinx-apidoc -o docs/ disruption_generator
```
Comment (jan-zmeskal) [inline Makefile]: I don't think we are ready to have auto-generated API docs. They actually look like mess and there's not much of them anyway. We can easily re-activate this later once we actually have some docstrings.

File: disruption_generator/cli.py [hunk 3]
```diff
@@ -39,6 +39,12 @@ def main(experiments_path):
 
 
 async def execute(experiments_path):
+    """Find, parse and execute experiments.
+
+    :param experiments_path: Path to directory with definitions of experiments.
+    :return: 0 on success, else 1.
+    :rtype: int
+    """
```
Comment (jan-zmeskal) [inline disruption_generator/cli.py]: This is just an example of how docstring should look like for autodoc make proper use of it.

File: docs/make.bat [hunk 4]
```diff
@@ -1,36 +0,0 @@
...
-)
-
-%SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS%
-goto end
-
-:help
-%SPHINXBUILD% -M help %SOURCEDIR% %BUILDDIR% %SPHINXOPTS%
-
-:end
-popd
```
Comment (jan-zmeskal) [inline docs/make.bat]: I don't believe we really support Windows setup at this moment, so why even have this?

File: docs/source/conf.py [hunk 5]
```diff
@@ -1,38 +1,45 @@
-#!/usr/bin/env python
```
Comment (jan-zmeskal) [inline docs/source/conf.py]: This has been built automatically by sphinx-quickstart, but I believe all the params are correct there.

**Review by rhrazdil — APPROVED** [review]
Nice.

**Comment by grafuls** [comment]
Try to build the docs from RTD first.

**Comment by jan-zmeskal** [comment]
The error message is quite funny: 

> Could not find a version that matches pluggy<0.7,<1.0,>=0.3.0,>=0.5,>=0.7

Well, pluggy probably won't be <0.7 and >=0.7 at the same time. :-) The wierd thing is that I cannot reproduce this locally. Any idea on this, @grafuls ?

**Comment by jan-zmeskal** [comment]
Hint by @grafuls: `pipenv graph`. I will look into this as soon as I clear some bugs out of my todo list.

**Comment by jan-zmeskal** [comment]
Okay, so I also fixed the problem with installing sub-dependencies. TL;DR it is pipenv's problem that must be worked around a little bit until it is properly solved. You can read about it [here](https://github.com/pypa/pipenv/issues/2596). If you want to see just the solution, check this [comment](https://github.com/pypa/pipenv/issues/2596#issuecomment-409365699) and let's say the next five comments.

The work-around is here: e8a4eeefae200ce7479dd2680849db85a42bb051

**Pull request merged**